news/4A0C40FF-ABE1-48C7-954C-7C3EB229135F.feature.rst

```diff
@@ -0,0 +1 @@
+Add ruff rules ASYNC,C4,C90,PERF,PLE,PLR for minor optimizations and to set upper limits on code complexity.
```

Comment:
This change would not cause a visible difference to the user and therefore should not appear in the changelog. This should use the `trivial` type instead.